src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -96,13 +96,17 @@ public void onBreedingGooey(BreedGooeyEvent event, EntityRef gooeyEntity, GooeyC
             String itemName = item.getComponent(DisplayNameComponent.class).name;
 
             if (!itemName.isEmpty() && hungerComponent.food.contains(itemName)) {
-                logger.info("Selected for breeding ...");
+                logger.info("Selected for breeding...");
                 FollowComponent followComponent = gooeyEntity.getComponent(FollowComponent.class);
                 BehaviorComponent behaviorComponent = gooeyEntity.getComponent(BehaviorComponent.class);
                 MatingComponent matingComponent = new MatingComponent();
 
                 followComponent.entityToFollow = event.getInstigator();
-                behaviorComponent.tree = assetManager.getAsset("GooKeeper:breedingBehavior", BehaviorTree.class).get();
+                if (!matingComponent.selectedForMating) {
```

Comment:
You could remove lines 105 to 109 and replace them with just `matingComponent.selectedForMating = !matingComponent.selectedForMating`